### PR TITLE
Guard against empty tasks list

### PR DESCRIPTION
**Pull request by jmsanders**
https://github.com/dagster-io/dagster/commit/cdc1f5a37487f4e9cf48a61115058537a7d2eb4b
guards against the describe_tasks response not having a "tasks" key.
This guards against the key's value being an empty list.

### CODE REVIEW

**Review by alangenfeld — APPROVED** [review]
Can we add some tests to prevent this from regressing in future implementation changes?

File: python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py [hunk 1]
```diff
@@ -137,11 +137,10 @@ def can_terminate(self, run_id):
...
         if arn and cluster:
-            status = (
-                self.ecs.describe_tasks(tasks=[arn], cluster=cluster)
-                .get("tasks", [{}])[0]
-                .get("lastStatus")
-            )
+            tasks = self.ecs.describe_tasks(tasks=[arn], cluster=cluster).get("tasks")
+            if not tasks:
+                return False
+            status = tasks[0].get("lastStatus")
```
Comment (rexledesma) [inline python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py:143]: maybe refactor this into a helper method `_get_task_for_run_id`?

**Review by rexledesma — APPROVED** [review]
This seems to be a pathological response from ecs - a comment or two could help with understanding in the future

**Comment by jmsanders** [comment]
@alangenfeld tests added - I think https://github.com/dagster-io/dagster/pull/4428/commits/b1d576109e323fd26a582c51e8b08737b31b5fcb and https://github.com/dagster-io/dagster/pull/4428/commits/d5ff25ebbdbf1430c8ca2d48a8128cf665087e12 might be a little overly defensive but 🤷

File: python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py [hunk 2]
```diff
@@ -123,45 +133,40 @@ def launch_run(self, context: LaunchRunContext) -> None:
...
             if status and status != "STOPPED":
                 return True
 
         return False
```
Comment (rexledesma) [inline python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py:154]: very small nit: this conditional has the main of the logic of this function. would prefer that it is deindented.

```suggestion
    if not (arn and cluster):
        return False
    tasks = self.ecs.describe_tasks(tasks=[arn], cluster=cluster).get("tasks")
    if not tasks:
        return False
    status = tasks[0].get("lastStatus")
    if status and status != "STOPPED":
        return True
```

**Review by rexledesma — APPROVED** [review]

**Pull request merged**